R/summarize_weighted_production.R

```diff
@@ -1,50 +1,101 @@
-#' Summarize production based on the weight of each loan per sector per year
+#' Summaries based on the weight of each loan per sector per year
 #'
-#' Summarize production based on the weight of each loan per sector per year.
+#' Based on the based on the weight of each loan per sector per year,
+#' `summarize_weighted_production()` and `summarize_weighted_buildout()`
+#' summarize the production and build-out, respectively.
 #'
 #' @param data A data frame like the output of [join_ald_scenario()].
 #' @param use_credit_limit Logical vector of length 1. `FALSE` defaults to using
-#'   the column `loan_size_outstanding`. Set to `TRUE` to use the column
-#'   `loan_size_credit_limit` instead.
+#'   the column `loan_size_outstanding`. Set to `TRUE` to instead use the column
+#'   `loan_size_credit_limit`.
 #' @param ... Variables to group by.
 #'
 #' @seealso [join_ald_scenario()].
 #'
 #' @export
 #'
+#' @section Warning:
+#' The build-out analysis excludes companies with 0 production. Build-out is
+#' undefined for companies that have no initial production; including such
+#' companies would cause build-out percentage to be infinite, which is wrong.
+#'
 #' @family utility functions
 #'
 #' @return A tibble with the same groups as the input (if any) and columns:
-#'   `sector`, `technology`, `year`, and `weighted_production`.
+#'   `sector`, `technology`, and `year`; and `weighted_production` or
+#'   `weighted_production` for `summarize_weighted_production()` and
+#'   `summarize_weighted_buildout()`, respectively.
 #'
 #' @examples
-#' library(r2dii.analysis)
+#' installed <- requireNamespace("r2dii.data", quietly = TRUE) &&
+#'   requireNamespace("r2dii.match", quietly = TRUE)
+#' if (!installed) stop("Please install r2dii.match and r2dii.data")
+#'
 #' library(r2dii.data)
 #' library(r2dii.match)
 #'
-#' master <- r2dii.data::loanbook_demo %>%
-#'   r2dii.match::match_name(r2dii.data::ald_demo) %>%
-#'   r2dii.match::prioritize() %>%
-#'   join_ald_scenario(r2dii.data::ald_demo,
-#'     r2dii.data::scenario_demo_2020,
+#' master <- loanbook_demo %>%
+#'   match_name(ald_demo) %>%
+#'   prioritize() %>%
+#'   join_ald_scenario(
+#'     ald = ald_demo,
+#'     scenario = scenario_demo_2020,
 #'     region_isos = region_isos_demo
 #'   )
```

Comment:
So we don't need it in `summarize_weighted_production()` nor in `summarize_weighted_percent_change()`?
If so, so we need `join_ald_scenario()` at all?

In the file that contains both functions I see just this line that uses `.data$scenario`:

```r
    group_by(.data$sector, .data$year, .data$scenario) %>%
```

However, removing `data$scenario` here makes the regression tests fail. So it changes the output. This may be okay, but I need you here.

--

For now I think your idea is the least disruptive; so I also prefer (1): That should simplify the implementation while preserving the interface.

Reply:
So it's a bit confusing. In principle, `join_ald_scenario` isn't necessary at all to calculate the summaries, so it is not necessary for `summmarize_weighted_production`. However, it is necessary for `target_*`, which call `summarize_weighted_production`. 

I have an idea for how we can extract these two things, but I propose we handle that in a separate issue/ pr

Reply:
See #149